### DISCUSSION TITLE
Need two '/' for getting data

### DISCUSSION THREAD

**Discussion by YunusovYF** [fastapi/fastapi#6349] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
router = APIRouter(
    prefix='/consumers',
    tags=['Consumers'],
    dependencies=[Depends(get_current_user)]
)

@router.get("/users/whoami", response_model=UserOutSchemaWithMsgOne)
def who_am_i(current_user: UserOutSchema = Depends(get_current_user)):
    logging.info(f'Запрос на информацию об авторизовавшемся пользователе {current_user = }')
    content, msg_type, msg = current_user, 'a', f'Done'
    return {'content': content, 'msg_type': msg_type, 'msg': msg}

@router.get("/userswhoami", response_model=UserOutSchemaWithMsgOne)
def who_am_i_test(current_user: UserOutSchema = Depends(get_current_user)):
    logging.info(f'Запрос на информацию об авторизовавшемся пользователе {current_user = }')
    content, msg_type, msg = current_user, 'a', f'Done'
    return {'content': content, 'msg_type': msg_type, 'msg': msg}
```

### Description

Today I found one problem and the search did not give me an answer. If you use two slashes "/users/whoami", the request is processed successfully.

![изображение](https://user-images.githubusercontent.com/94920021/159635034-bc375833-0cc6-4fed-80cf-c3b973d7c2c2.png)

If use one slash "/userswhoami", then it gives an error, the code is completely identical.

![изображение](https://user-images.githubusercontent.com/94920021/159635070-f48d9b90-8521-401d-b162-b29adfc0305b.png)

### Operating System

Linux

### Operating System Details

Ubuntu 20.04

### FastAPI Version

75.0

### Python Version

3.8

**Comment by TheoBabilon** [comment, marked as answer]
I'm guessing (and this is the only explanation that makes sense to me) you certainly have another route defined BEFORE (order matters!!) that "/userswhoami" route, and it must be accepting an integer x - so the route may look like it: "/{x}"

so when you're trying to access "/userswhoami" this integer route is used by FastAPI instead of the "/userwhoami" route - and because "userswhoami" is not an integer you're getting an error

**Comment by YunusovYF** [comment]
@TheoBabilon thank you!
